src/main/groovy/org/labkey/gradle/util/BuildUtils.groovy

```diff
@@ -964,4 +973,22 @@ class BuildUtils
             project.logger.debug("No ${configName} configuration found for ${project.path}.")
         }
     }
+
+    enum BuildFromSource {
+        _TRUE,
+        _FALSE,
+        FORCE
+
+        static BuildFromSource fromProperty(Project project, String propertyName) {
+            String propertyValue = project.hasProperty(propertyName) ? project.property(propertyName) : null
+            fromProperty(propertyValue)
+        }
+
+        static BuildFromSource fromProperty(String propertyValue) {
+            if ('force' == propertyValue)
+                return FORCE
+            else
+                return Boolean.valueOf(propertyValue) ? _TRUE : _FALSE
```

Comment:
This is changing the behavior of the current property. Currently, the property has to be explicitly set to false to turn off building from source. Otherwise, if the property is there or set to true we will build from source.

Reply:
Maybe I'm missing something. The only place that currently checks the value of `buildFromSource` is in `BuildUtils.whyNotBuildFromSource`. It also uses `Boolean.valueOf()` which only returns `true` for `value.equalsIgnoreCase("true")`:
```
else if (!Boolean.valueOf((String) value))
    reasons.add("${property} property is false")
```
It has been updated to use this enum and should behave the same way:
```
else if (BuildFromSource.fromProperty(value) == BuildFromSource._FALSE)
    reasons.add("${property} property is not 'true'")
```

Reply:
Ah you are right. Too much Javascript in my brain.